BCFishNet/Assets/BCScripts/BC/BCManager.cs

```diff
@@ -0,0 +1,286 @@
+    }
+
+    private string _playerName;
+    public string PlayerName
+    {
+        get => _playerName;
+        set => _playerName = value;
+    }
+    private string _externalId;
+    public string ExternalId
+    {
+        get => _externalId;
+        set => _externalId = value;
+    }
+
+    private void OnAuthSuccess(string responseData, object cbObject, Action<bool> callback)
+    {
+        Debug.Log("Authentication successful");
+
+        try
+        {
+            var response = JsonReader.Deserialize<Dictionary<string, object>>(responseData);
+
+            if (response.TryGetValue("data", out object dataObj) &&
+                dataObj is Dictionary<string, object> data &&
+                data.TryGetValue("playerName", out object playerNameObj))
+            {
+                _playerName = playerNameObj?.ToString();
+                Debug.Log($"Stored playerName: {_playerName}");
+            }
+        }
+        catch (Exception ex)
+        {
+            Debug.LogWarning($"Failed to parse playerName from response: {ex.Message}");
+        }
+
+        callback?.Invoke(true);
+    }
+
+    private void OnAuthFailure(int error, int code, string statusMessage, object cbObject, Action<bool> callback)
+    {
+        Debug.LogWarning($"Authentication failed: {statusMessage} (code: {code}, error: {error})");
+        callback?.Invoke(false);
+    }
+
+    public void AuthenticateAnonymous(Action<bool> callback)
+    {
+        SuccessCallback success = (responseData, cbObject) =>
+        {
+            OnAuthSuccess(responseData, cbObject, callback);
+        };
+
+        FailureCallback failure = (error, code, statusMessage, cbObject) =>
+            OnAuthFailure(error, code, statusMessage, cbObject, callback);
+
+        _bc.AuthenticateAnonymous(success, failure);
+    }
+
+    public void AuthenticateUser(string username, string password, Action<bool> callback)
+    {
+        SuccessCallback success = (responseData, cbObject) =>
+            OnAuthSuccess(responseData, cbObject, callback);
+
+        FailureCallback failure = (statusMessage, code, error, cbObject) =>
+            OnAuthFailure(statusMessage, code, error, cbObject, callback);
+
+        _bc.AuthenticateUniversal(username, password, true, success, failure);
+    }
+
+    public void EnableRTT(bool enabled, Action OnSuccess)
+    {
+        if (enabled)
+        {
+            SuccessCallback success = (responseData, cbObject) =>
+            {
+                OnSuccess();
+            };
+            _bc.RTTService.EnableRTT(success, OnRTTFailed);
+        }
+        else
+        {
+            _bc.RTTService.DisableRTT();
+        }
+    }
+
+    private void OnRTTFailed(int statusCode, int reasonCode, string statusMessage, object cbObject)
+    {
+        Debug.LogError($"RTT disconnected: {statusMessage} (Status Code: {statusCode}, Reason Code: {reasonCode})");
+        // Error handling logic
```

Comment:
Is there a catch all error handling in place? Should that be added before publishing the example?

Reply:
Yes! Thanks for catching it! 

I will handle something such that while in game if it fails, it is booted to the main menu with a message saying disconnected, and reattempting to connect automatically. Since we connect RTT automatically after authentication.